### ISSUE TITLE
No value defined for temporary

### ISSUE THREAD

**Issue by cgay** [dylan-lang/opendylan#1523]
Reproducer: https://play.opendylan.org/shared/d085b8e00fe79ab9

```dylan
iterate loop (a = #f, b = 1)
  if (b == 3)
    b
  else
    loop(b, b + 1)
  end
end
```

The above compiles without error if `loop(b, b + 1)` is changed to `loop(a, b + 1)`.

**Comment by housel** [comment]
This might be a duplicate of #1461.

**Comment by housel** [comment]
Fixed by #1622.
